pilot/pkg/model/eventhandler.go

```diff
@@ -0,0 +1,20 @@
+// Copyright 2020 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model
+
+type DistributionEventHandler interface {
```

Comment:
I think we already have a big problem with the InstanceHandler ( which is no longer used in practice and very inefficient ) and all the complexity around. 

The events are specific to the XDS implementation and behavior - we shouldn't add another over-generic interface.

Reply:
The reason for the interface is to prevent a circular list of dependencies in which status depends on model and model depends on status.  If there is a better place for this interface which does not create a circular dependency, I'd be happy to move it.

Reply:
model should not depend on Status I think

Reply:
Agreed.  That is the purpose of this interface.  If I get rid of the interface, ads.go will be calling the status/Reporter directly, which would create that undesirable dependency loop.

Reply:
This interface has been refactored into the v2 package.  Resolving this thread unless you think further action is merited.